cli/src/pcluster/models/imagebuilder_config.py

```diff
@@ -185,3 +193,85 @@ def _set_default(self, dev_settings: ImagebuilderDevSettings):
         for key, value in self.__dict__.items():
             if not key.startswith("_") and key not in self._cfncluster_attributes:
                 self._cfncluster_attributes.update({key: value})
+
+
+class ImageBuilderBucket(S3Bucket):
```

Comment:
do we really need two different classes to manage the cluster and the image builder bucket? from a quick look it looks like they perform the same operations but on different directories. probably I'm oversimplifying but aren't these classes different only on the value of the attributes that can be assigned when instantiated?

Reply:
but you could use a factory that is responsible to instantiate such classes in the proper way. Also _get_cluster_dir and _get_image_dir in the end are just returning the base path where to upload the files and they don't have to contain cluster or image in the name.
my concern is that there is some good amount of duplicated code. At the very least if we don't want to get rid of this separation let's try to generalize all the functinalities and put them in the base class and then have the subclasses only assign parameters about the various dirs.

In the end we want to have a generic object that manages the S3 bucket and exposes a set of operations to upload different types of files